CockpitPlugin/src/main/java/com/polidea/cockpitplugin/generator/BaseCockpitGenerator.kt

```diff
@@ -64,21 +64,21 @@ abstract class BaseCockpitGenerator {
     }
 
     protected fun mapToTypeClass(param: Param<*>): Class<*> {
-        return when (param) {
-            is BooleanParam -> Boolean::class.java
-            is DoubleParam -> Double::class.java
-            is IntegerParam -> Int::class.java
-            is StringParam -> String::class.java
+        return when (param.value) {
+            is Boolean -> Boolean::class.java
+            is Double -> Double::class.java
+            is Int -> Int::class.java
+            is String -> String::class.java
             else -> throw IllegalArgumentException("Param type undefined: $param!")
         }
     }
 
     protected fun mapToJavaObjectTypeClass(param: Param<*>): Class<*> {
-        return when (param) {
```

Comment:
`when` expressions is not needed any more. We can simplify it to `param.value::class.javaObjectType`